cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -761,6 +762,64 @@ public static KafkaCluster fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup vers
         return result;
     }
 
+    /**
+     * Checks that the Cruise Control metrics topic replication level is above the cluster's minimum insync replication value. If it is less than or
+     * equal to the minISR then it will be set to minISR + 1, unless that value is greater than the number of brokers in the cluster. If that is the
+     * case then it is set to the number of brokers provided that is greater than minISR, if not then a warning is sent and it is set as minISR. If the
+     * configured CC metric replication is greater than minISR + 1 then it is left as is.
+     *
+     * @param configuration
+     * @param result
+     * @param kafkaClusterSpec
+     * @return
+     */
+    private static String checkCCMetricsReplication(KafkaConfiguration configuration, KafkaCluster result, KafkaClusterSpec kafkaClusterSpec) {
+
+        int configuredCCReplicationFactor;
+        if (configuration.getConfigOption(CC_REPLICATION_FACTOR_CONFIG_FIELD) == null) {
+            configuredCCReplicationFactor = Integer.parseInt(result.kafkaDefaultReplicationFactor);
+        } else {
+            configuredCCReplicationFactor = Integer.parseInt(configuration.getConfigOption(CC_REPLICATION_FACTOR_CONFIG_FIELD));
+        }
+        int brokerMinInsyncReplicas;
+        if (configuration.getConfigOption(KAFKA_MIN_INSYNC_REPLICA_CONFIG_FIELD) == null) {
+            brokerMinInsyncReplicas = Integer.parseInt(result.kafkaDefaultMinInsyncReplias);
+        } else {
+            brokerMinInsyncReplicas = Integer.parseInt(configuration.getConfigOption(KAFKA_MIN_INSYNC_REPLICA_CONFIG_FIELD));
+        }
+
+        // We want the CC metrics topics to have a replication factor at least one higher than the broker minimum so that metrics
+        // producer acks are likely to be acknowledged faster.
+        int checkedCcReplicationFactor = Math.max(configuredCCReplicationFactor, brokerMinInsyncReplicas + 1);
```

Comment:
are we really sure that the +1 is useful? because due to this addition you have the following code to check that it doesn't go over the replicas.